documentation/src/main/asciidoc/ch11.asciidoc

```diff
@@ -694,17 +694,25 @@ mechanism to register additional constraint definitions. All you have to do is t
 _javax.validation.ConstraintValidator_ to _META-INF/services_. In this service file you list the
 fully qualified classnames of your constraint validator classes (one per line). Hibernate Validator
 will automatically infer the constraint types they apply to.
-See <<example-using-service-file-for-constraint-definitions>> for an example.
+See <<example-using-service-file-for-constraint-definitions,Constraint definition via service file>>
+for an example.
 
 [[example-using-service-file-for-constraint-definitions]]
-.Using _META-INF/services/javax.validation.ConstraintValidator_ for constraint definitions
+._META-INF/services/javax.validation.ConstraintValidator_
+
 ====
 [source]
 ----
+# Assuming a custom constraint annotation @org.mycompany.CheckCase
 org.mycompany.CheckCaseValidator
 ----
 ====
 
+To contribute default messages for your custom constraints, place a _ValidationMessages.properties_
```

Comment:
> Simple, no need for yet another contract/api

API? It'd be a defined bundle name. I guess you could consider that as a contract.

> This makes the connection to Bean Validation obvious. It also prevent proliferation of names when everyone tries to name the properties file something else

Actually it's re-using the defined name from the spec with slightly other semantics I'm a bit concerned about. A separate name would make more apparent that this is a HV-specific feature. Re everybody using different names, it'd be one single defined name.

> Using a new contract, people could for example put a ValidationMessages.properties into the root of their jar. This would then conflict with the users ValidationMessages.properties

Right, it'd have to be another name for that reason.

> Don't you think this rule is pretty straight forward?

I guess it is. Still I find re-using the BV-defined bundle name not optimal.